flink-libraries/flink-table/src/main/scala/org/apache/flink/table/functions/utils/UserDefinedFunctionUtils.scala

```diff
@@ -320,12 +348,20 @@ object UserDefinedFunctionUtils {
     } else if(extractedType != null) {
       extractedType
     } else {
-      TypeExtractor
+      try {
+        TypeExtractor
```

Comment:
I think it is more safe than `TypeInformation.of(Any)`, because `TypeInformation.of(Any)` can only works for non-generic types. 

For example: 

```
public class UDAF extends AggregateFunction[Tuple2<String, Long>, Long] {
    public Long createAccumulator() {...}
    public Tuple2<String, Long> getValue(Long acc) {...}
}
```
For the given UDAF, the return type is a generic type `Tuple2<String, Long>`. We can't extract the return type  from an instance of Tuple2, because of Java type erasure. But we can extract it from the type hierarchy of UDAF class. 

Am I right, @twalthr ?



```